### PR TITLE
Support ingesting parsable bytestreams

### PR DESCRIPTION
**Pull request by wemrysi**
[ch2253]
[ch656]

### CODE REVIEW

File: frontend/src/test/scala/quasar/ParseInstructionSpec.scala [hunk 1]
```diff
@@ -569,6 +569,18 @@ object ParseInstructionSpec {
...
+      "compact array containing nested arrays with single nested value retained" in {
+        val input = ldjson("""
+          { "a": [[[1, 3, 5], "k"], "foo", { "b": [5, 6, 7], "c": [] }], "d": "x" }
+          """)
+
+        val expected = ldjson("""
+          { "a": [{"b": [5, 6, 7] }] }
+          """)
+
+        input must maskInto(".a[2].b" -> Set(Array))(expected)
```
[the remark this answers is not in the document]
Comment (wemrysi) [reply frontend/src/test/scala/quasar/ParseInstructionSpec.scala]: Yeah, it definitely had to do with multiple nested arrays, my first attempt at the test didn't catch it either, required moar nesting.

[the remark this answers is not in the document]
Comment (djspiewak) [reply frontend/src/test/scala/quasar/ParseInstructionSpec.scala]: 💪

File: impl/src/main/scala/quasar/impl/datasource/local/LocalDatasource.scala [hunk 2]
```diff
@@ -54,31 +50,27 @@ final class LocalDatasource[F[_]: ContextShift: Timer] private (
...
+      jp <- toNio(path)
+
+      exists <- F.delay(Files.exists(jp))
+      _ <- exists.unlessM(RE.raiseError(pathNotFound(path)))
+
+      isFile <- F.delay(Files.isRegularFile(jp))
+      _ <- isFile.unlessM(RE.raiseError(notAResource(path)))
+    } yield {
+      QueryResult.typed(
+        ParsableType.json(JsonVariant.LineDelimited, true),
```
[the remark this answers is not in the document]
Comment (wemrysi) [reply impl/src/main/scala/quasar/impl/datasource/local/LocalDatasource.scala]: Aye, I was actually thinking we could optionally do this generally for soon-to-be-introduced `Untyped` `QueryResult` if no other type is specified.

File: connector/src/main/scala/quasar/connector/QueryResult.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import qdata.QDataDecode
+import monocle.Prism
+import scalaz.{~>, Cord, Show}
+import scalaz.syntax.show._
+import shims._
+
+sealed trait QueryResult[F[_]] extends Product with Serializable {
+  import QueryResult._
+
+  def data: Stream[F, _]
```
[the remark this answers is not in the document]
Comment (wemrysi) [reply connector/src/main/scala/quasar/connector/QueryResult.scala]: It _seemed_ like it'd be a bit nicer, as we'll either need extra tparams or carry around a `QueryResult[F, _]`, but we'll see how it ends up. We can always swap to what you describe whilst adding the `Untyped` case if it makes more sense.

**Pull request merged**